File: day17/spinlock.py
#!/usr/bin/env python3
from sys import argv

stride = int(argv[1])
MAX = 2017
buf = [0]
p = 0

for i in range(1, MAX+1):
    p = (p + stride) % len(buf)
    buf.insert(p+1, i)
    p += 1

for i in range(len(buf)):
    if buf[i] == 2017:
        j = (i + 1) % len(buf)
        print("result1:", buf[j])
        break

File: day17/spinlock.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);

my $stride = $ARGV[0];

my @buf = (0);
my $p = 0;

for my $i (1..2017) {
    $p = ($p + $stride) % @buf;
    splice @buf, $p+1, 0, $i;
    $p++;
}

for my $i (0..$#buf) {
    if ($buf[$i] == 2017) {
        my $j = ($i+1) % @buf;
        say "result1: $buf[$j]";
        last;
    }
}
